### PR TITLE
feat(bivariate-matrix): 11602 - preselected layers from presets on ma…

### PR DESCRIPTION
**Pull request by propakov**
https://kontur.fibery.io/Tasks/Task/Preselected-layers-from-presets-on-matrix-11602
Screenshot 2022-09-14 at 14 19 14

### CODE REVIEW

File: src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts [hunk 1]
```diff
@@ -10,24 +12,45 @@ import { bivariateNumeratorsAtom } from '~features/bivariate_manager/atoms/bivar
...
-      yDenominator: string | null,
-    ) => ({ xNumerator, xDenominator, yNumerator, yDenominator }),
+    enabledLayersAtom,
+    layersLegendsAtom,
+
+    calculateSelectedCell: () => undefined,
+    setPreselectMode: (flag: boolean) => flag,
+    runPreselection: () => undefined,
+    setMatrixSelection: formatSelection,
+    presetMatrixSelection: (selection: SelectionInput) => selection,
```
Comment (Akiyamka) [inline src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts]:
```suggestion
    calculateSelectedCell: () => null,
    setPreselectMode: (flag: boolean) => flag,
    runPreselection: () => null,
    setMatrixSelection: formatSelection,
    presetMatrixSelection: (selection: SelectionInput) => selection,
```

File: src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts [hunk 2]
```diff
@@ -10,24 +12,49 @@ import { bivariateNumeratorsAtom } from '~features/bivariate_manager/atoms/bivar
...
+const DEFAULT_STATE = {
   xNumerator: null,
   xDenominator: null,
   yNumerator: null,
   yDenominator: null,
+
+  preselectMode: false,
+  selectedCell: null,
+  selectCellCallback: null,
 };
```
Comment (IlyaIzr) [inline src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts]: Shouldn't `DEFAULT_STATE` be connected with `type SelectionInput` somehow?

File: src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts [hunk 3]
```diff
@@ -38,10 +65,172 @@ export const bivariateMatrixSelectionAtom = createAtom(
...
+        });
+      }
+    });
+
+    // this is onSelect method from BivariateMatrixControl
+    onAction('setSelectCellCallback', (selectCellCallback) => {
+      state = { ...state, selectCellCallback };
+    });
+
+    const getActiveBivariateLayer = () => {
```
Comment (IlyaIzr) [inline src/features/bivariate_manager/atoms/bivariateMatrixSelection.ts]: `getMounterBivariateLayer` or rather `getEnabledBivariateLayer` would be more consistant

**Review by IlyaIzr — APPROVED** [review]
Aprroved with some comments, but I'd like @Akiyamka to approve this one too as I was envolved in a development a bit and can't have a clear look.

Comments are really helpful!

**Pull request merged**